src/System.IO.Ports/src/System/IO/Ports/InternalResources.cs

```diff
@@ -77,7 +77,10 @@ internal static void WinIOError(int errorCode, String str)
                         throw new UnauthorizedAccessException(string.Format(SR.UnauthorizedAccess_IODenied_Path, str));
 
                 case Interop.Errors.ERROR_FILENAME_EXCED_RANGE:
-                    throw new PathTooLongException(SR.IO_PathTooLong);
+                    if (string.IsNullOrEmpty(str))
```

Comment:
Is it actually possible for str to be null or empty here?

Reply:
I can't think why path too long would be appropriate if the path was empty or null?

Reply:
> Is it actually possible for str to be null or empty here?

Well, I do not know this. But for all other cases there (path not found, access denied...) there is a check. Though the checks are only for empy string (`str.Length == 0`). The second thing is, that it is this way also in coreclr, as @jkotas pointed in his review. Though it is in `System/IO/Win32Marshal.cs`.

> I can't think why path too long would be appropriate if the path was empty or null?

So I am not sure what to do here. If to leave it as I did it, or change it to test only for empty string like the other cases, or revert it back. Personally, I would remove the check for empty stirng and left only new message with path. And I would make it the same way in `Win32Marshal.cs` file.

Reply:
> Is it actually possible for str to be null or empty here?

It is possible when the OS API returns the path too long error code, but the caller of this method does not have the path.